documentation/src/docs/streamcuts.md

```diff
@@ -65,30 +65,38 @@ ReaderGroupConfig.builder()
                 .build();
 ```
 
-The below API can be used to reset an existing ReaderGroup with a new ReaderGroup configuration instead creating a
-ReaderGroup.
+The below API can be used to **reset an existing Reader Group with a new Reader Group** configuration instead creating a
+Reader Group.
 ```
 /*
  * ReaderGroup API used to reset a ReaderGroup to a newer ReaderGroup configuration.
  */
 io.pravega.client.stream.ReaderGroup.resetReaderGroup(ReaderGroupConfig config)
 ```
+The API `io.pravega.client.stream.ReaderGroup#generateStreamCuts`, generates a `StreamCut` after co-ordinating with all the Readers using `io.pravega.client.state.StateSynchronizer`. A `StreamCut` is generated by using the latest Stream Segment read offsets returned by the Readers along with unassigned segments (if any). The configuration `ReaderGroupConfig.getGroupRefreshTimeMillis()` decides the maximum delay by which the Readers return the latest read offsets of their assigned segments.
+
+The `StreamCut` generated by this API can be used by the application as a reference to a Position in the stream. This is guaranteed to be greater than or equal to the position of the Readers at the point of invocation of the API.
+
 ## StreamCut with BatchClient
 
-`StreamCut` representing the current head and current tail of a stream can be obtained using below BatchClient API.
+`StreamCut` representing the current head and current tail of a stream can be obtained using below `BatchClient` API.
```

Comment:
This should be removed, as the api to fetch the current `TAIL and HEAD` of the stream is  on `StreamManager` and not `BatchClient`.

Perhaps we could add a section which talks about obtaining `HEAD` and `TAIL` streamcuts for a Stream. In the batch client section should contain how it is used as in line 88.